### PR TITLE
feat(TDI-40684) : Support snowflake on Azure

### PR DESCRIPTION
**Pull request by ypiel-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-40662

**What is the new behavior?**
We can now select the provider 'AWS' or 'AZURE'. If azure is selected we can select the region.

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/103/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/103/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aypiel%2FTDI-40684_snowflake_on_azure)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/104/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/104/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aypiel%2FTDI-40684_snowflake_on_azure)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/106/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/106/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aypiel%2FTDI-40684_snowflake_on_azure)

File: components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java [hunk 1]
```diff
@@ -253,8 +287,24 @@ public String getConnectionUrl() {
...
+        }
+        String url = new StringBuilder().append("jdbc:snowflake://")
+                .append(account)
+                .append(azure)
+                .append(".snowflakecomputing.com")
+                .append("/?")
+                .append(stringBuilder)
+                .toString();
+
+        if (LOGGER.isDebugEnabled())
```
Comment (undx) [inline components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java]: Add braces please.

Comment (undx) [reply components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java]: Usually, we do don't as in javajets (if (LOG.isXXX). Just log.

File: components/components-snowflake/components-snowflake-definition/src/main/resources/org/talend/components/snowflake/SnowflakeConnectionProperties.properties [hunk 2]
```diff
@@ -21,4 +23,59 @@ form.Reference.title=Connection Reference
 form.Reference.displayName=Connection Reference
 
 messages.connectionSuccessful=Connection successful
-error.missingAccount=Missing account
\ No newline at end of file
+error.missingAccount=Missing account
+
+property.possiblevalue.BRAZIL_SOUTH.displayName=Americas - Brazil South
```
Comment (undx) [inline components/components-snowflake/components-snowflake-definition/src/main/resources/org/talend/components/snowflake/SnowflakeConnectionProperties.properties]: Are these labels visibles ? Maybe should be placed in `AzureRegion.properties` or package's `Messages.properties` ?

File: components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java (hunk 1 above)
Comment (ypiel-talend) [reply components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java]: just don't want to concat string if debug is disabled

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/107/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/107/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aypiel%2FTDI-40684_snowflake_on_azure)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/163/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/163/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/5/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/5/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/23/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/23/)

**Comment by ypiel-talend** [comment]
Fail on ci with error : "Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.test.web.client.TestRestTemplate': Initialization of bean failed; nested exception is java.lang.NoClassDefFoundError: org/apache/http/protocol/HttpContext"
Success on local. I merge.

**Pull request merged**